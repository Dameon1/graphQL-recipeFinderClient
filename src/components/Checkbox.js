import React from "react";
import { IngredientBox } from "./styles";

export class Checkbox extends React.Component {
  state = {
    isChecked: false
  };

  toggleCheckboxChange = () => {
    const { handleCheckboxChange, label } = this.props;
    this.setState(({ isChecked }) => ({ isChecked: !isChecked }));
    handleCheckboxChange(label);
  };

  render() {
    const { label } = this.props;
    const { isChecked } = this.state;

    return (
      <IngredientBox onClick={this.toggleCheckboxChange}>
        <label>
          <input
            type="checkbox"
            value={label}
            checked={isChecked}
            onChange={this.toggleCheckboxChange}
          />
          {label}
        </label>
      </IngredientBox>
    );
  }
}

export default Checkbox;