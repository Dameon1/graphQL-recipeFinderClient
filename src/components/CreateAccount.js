import React from "react";
import { Mutation } from "react-apollo";
import { navigate } from "@reach/router";

import Error from "./ErrorMessage";
import { CURRENT_USER_QUERY, SIGNUP_MUTATION } from "../actions";
import { Form, UserSignIn, SignInButton, SignInInput, Label } from "./styles";

function routeToHome() {
  navigate("/dashboard");
}

export class CreateAccount extends React.Component {
  state = {
    password: "",
    username: ""
  };
  saveToState = e => {
    this.setState({ [e.target.name]: e.target.value });
  };
  render() {
    return (
      <Mutation
        mutation={SIGNUP_MUTATION}
        variables={this.state}
        refetchQueries={[{ query: CURRENT_USER_QUERY }]}
      >
        {(signup, { data, error, loading }) => {
          if (data && data.createUser && data.createUser.username) {
            routeToHome();
          }
          return (
            <UserSignIn>
              <Form
                method="post"
                onSubmit={async e => {
                  e.preventDefault();
                  await signup();
                  this.setState({ password: "", username: "" });
                }}
              >
                <Error error={error} />

                <Label htmlFor="username">Username</Label>
                {data && data.createUser.message && (
                  <Error error={data.createUser} />
                )}
                <SignInInput
                  name="username"
                  placeholder="username"
                  value={this.state.username}
                  onChange={this.saveToState}
                  type="text"
                />
                <Label htmlFor="password">Password</Label>
                <SignInInput
                  type="password"
                  name="password"
                  placeholder="password"
                  value={this.state.password}
                  onChange={this.saveToState}
                />
                <SignInButton
                  type="submit"
                  disabled={this.props.pristine || this.props.submitting}
                >
                  Register
                </SignInButton>
              </Form>
            </UserSignIn>
          );
        }}
      </Mutation>
    );
  }
}

export default CreateAccount;
