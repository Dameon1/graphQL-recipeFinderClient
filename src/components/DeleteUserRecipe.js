import React from "react";
import { Mutation } from "react-apollo";
import PropTypes from "prop-types";
import { DELETE_USER_RECIPE_MUTATION } from '../actions';

const DeleteUserRecipe = props => {
  return (
    <Mutation {...props} mutation={DELETE_USER_RECIPE_MUTATION} variables={props.variables}>
      {payload => props.children(payload)}
    </Mutation>
  );
};

DeleteUserRecipe.propTypes = {
  children: PropTypes.func.isRequired
};

export default DeleteUserRecipe;