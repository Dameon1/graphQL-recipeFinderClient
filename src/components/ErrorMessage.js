import React from "react";
import { ErrorStyles } from "./styles";
import PropTypes from "prop-types";

const ErrorMessage = ({ error }) => {
  if (!error || !error.message) return null;
  if (
    error.networkError &&
    error.networkError.result &&
    error.networkError.result.errors.length
  ) {
    return error.networkError.result.errors.map((error, i) => (
      <ErrorStyles key={i}>
        <p data-test="graphql-error">
          <strong>Shoot!</strong>
          {error}
        </p>
      </ErrorStyles>
    ));
  }
  return (
    <ErrorStyles>
      <p data-test="graphql-error">
        <strong>Shoot!</strong>
        {error.message}
      </p>
    </ErrorStyles>
  );
};

ErrorMessage.defaultProps = {
  error: {}
};

ErrorMessage.propTypes = {
  error: PropTypes.object
};

export default ErrorMessage;
