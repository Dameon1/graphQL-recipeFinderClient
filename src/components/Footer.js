//TODO ** CREATE STYLED COMPONENTS**

import React from "react";
import {FooterContainer,FooterText,FooterLink} from './styles';
import "./styles/footer.css";

export default function Footer() {
  return (
    <FooterContainer>
      <FooterText>
        Powered by{" "}
        <FooterLink
          href="https://spoonacular.com/"
          target="blank"
        >
          @Spoonacular
        </FooterLink>
      </FooterText>
    </FooterContainer>
  );
}