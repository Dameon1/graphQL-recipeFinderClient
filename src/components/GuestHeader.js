import React from "react";
import { Link } from "@reach/router";
import { NavButton } from "./styles";

export default class GuestHeader extends React.Component {
  render() {
    return (
      <nav>
        <Link to="/signIn">
          <NavButton>Login</NavButton>
        </Link>
        <Link to="/signUp">
          <NavButton>Sign up</NavButton>
        </Link>
        <Link to="/dashboard">
          <NavButton>Dashboard</NavButton>
        </Link>
        <Link to="/">
          <NavButton>Home</NavButton>
        </Link>
      </nav>
    );
  }
}
