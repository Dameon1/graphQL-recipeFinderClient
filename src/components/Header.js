import React from "react";
import { Link } from '@reach/router';
import User from './User';
import './styles/header.css';
import Spinner from 'react-spinkit';
import SignOut from './SignOut';
// import { signUserOut } from '../../actions/auth';
// import { signingUserOut } from "../../actions/userActions";

 export default class Header extends React.Component {

//   handleSignOut() {
//    this.props.dispatch(signingUserOut());
//    this.props.dispatch(signUserOut());
//   };

//TODO SEPARATE HEADERS INTO SEPARATE FILES
  render() {
    return (      
      <User>
        {(data) => {
          console.log(data.loading);
          if(data.loading) {
           return <Spinner spinnername="circle" fadeIn='none' />;
          }
          if(!data.data.me) {
            return (
              <div className='header'>
                <h1 className="headerText">
                  what2eat
                </h1>
                <div className="navLinks">
                  <Link to="/signIn" >
                    <button className="navLinkButtons">Login</button>
                  </Link>
                  <Link to="/signUp" >
                    <button className="navLinkButtons">Sign up</button>
                  </Link>  
                  <Link to="/dashboard" >
                    <button className="navLinkButtons" >Dashboard</button>
                  </Link>
                  <Link to="/" >
                    <button className="navLinkButtons">Home</button>
                  </Link>
                </div>
              </div>
            )
          } else {
            return (
              <div className='header'>
                <h1 className="headerText">
                  what2eat
                </h1>
                <div className="navLinks">
                  <Link to="/dashboard" >
                    <button className="navLinkButtons">Dashboard</button>
                  </Link>
                  <Link to="/myRecipes" >
                    <button className="navLinkButtons">My recipes</button>
                  </Link> 
                  <Link to="/" >
                    <SignOut />
                  </Link> 
                </div>
              </div>
          )
          }
        }}
      </User>
    )

    // if (!this.props.loggedIn) {
    //   return (
        
    // )}
     
  };
};
