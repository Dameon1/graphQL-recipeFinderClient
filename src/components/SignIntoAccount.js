import React from "react";
import { Mutation } from "react-apollo";
import { navigate } from "@reach/router";

import { Form, UserSignIn, SignInButton, SignInInput, Label } from "./styles";
import Error from "./ErrorMessage";

//TODO************ CREATE INDEX FOR ALL QUERY?MUTATIONS *********
// import CURRENT_USER from "../actions/currentUserQuery";
// import SIGNIN_MUTATION from "../actions/signInMutation";
import {CURRENT_USER_QUERY,SIGNIN_MUTATION} from '../actions';

function routeToHome() {
  navigate("/dashboard");
}

export class SignIntoAccount extends React.Component {
  state = {
    password: "",
    username: ""
  };
  saveToState = e => {
    this.setState({ [e.target.name]: e.target.value });
  };
  render() {
    return (
      <Mutation
        mutation={SIGNIN_MUTATION}
        variables={this.state}
        refetchQueries={[{ query: CURRENT_USER_QUERY }]}
      >
        {(signInUser, { data, error, loading }) => {
          if (data && data.signInUser && data.signInUser.username) {
            routeToHome(data);
          }
          return (
            <UserSignIn>
              <Form
                method="post"
                onSubmit={async e => {
                  e.preventDefault();
                  await signInUser();
                  this.setState({ password: "", username: "" });
                }}
              >
                <Error error={error} />
                <Label htmlFor="username">Username</Label>
                {data && data.signInUser.message && (
                  <Error error={data.signInUser} />
                )}
                <SignInInput
                  name="username"
                  placeholder="username"
                  value={this.state.username}
                  onChange={this.saveToState}
                  type="text"
                />
                <Label htmlFor="password">Password</Label>
                <SignInInput
                  type="password"
                  name="password"
                  placeholder="password"
                  value={this.state.password}
                  onChange={this.saveToState}
                />
                <SignInButton
                  type="submit"
                  disabled={this.props.pristine || this.props.submitting}
                >
                  Sign In
                </SignInButton>
              </Form>
            </UserSignIn>
          );
        }}
      </Mutation>
    );
  }
}

export default SignIntoAccount;
