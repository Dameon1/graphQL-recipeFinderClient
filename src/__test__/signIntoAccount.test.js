import React from "react";
import { shallow, mount } from "enzyme";
import { SignIntoAccount } from "../components/SignIntoAccount";
import toJSON from "enzyme-to-json";

//TODO: COULD ADD A FEW MORE SMOKE TESTS

describe("<SignIntoAccount />", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<SignIntoAccount />);
    //console.log(wrapper.debug());
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});