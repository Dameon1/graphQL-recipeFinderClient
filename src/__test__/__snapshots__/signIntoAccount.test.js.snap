// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIntoAccount /> renders without crashing 1`] = `
<Mutation
  mutation={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "SIGNIN_MUTATION",
          },
          "operation": "mutation",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "username",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "username",
                      },
                    },
                  },
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "password",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "password",
                      },
                    },
                  },
                ],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "signInUser",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "username",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "reason",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "message",
                      },
                      "selectionSet": undefined,
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [
            Object {
              "defaultValue": undefined,
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "username",
                },
              },
            },
            Object {
              "defaultValue": undefined,
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "password",
                },
              },
            },
          ],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 181,
        "start": 0,
      },
    }
  }
  refetchQueries={
    Array [
      Object {
        "query": Object {
          "definitions": Array [
            Object {
              "directives": Array [],
              "kind": "OperationDefinition",
              "name": Object {
                "kind": "Name",
                "value": "CURRENT_USER_QUERY",
              },
              "operation": "query",
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "me",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "username",
                          },
                          "selectionSet": undefined,
                        },
                      ],
                    },
                  },
                ],
              },
              "variableDefinitions": Array [],
            },
          ],
          "kind": "Document",
          "loc": Object {
            "end": 64,
            "start": 0,
          },
        },
      },
    ]
  }
  variables={
    Object {
      "password": "",
      "username": "",
    }
  }
>
  <Component />
</Mutation>
`;
